### PR TITLE
[Hub Generated] Review request for Microsoft.Cache to add version preview/2019-07-01

### PR DESCRIPTION
**Pull request by ghost**
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/sspeaks610/azure-rest-api-specs/dev-redis-Microsoft.Cache-2019-07-01...master/).

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “**WaitForARMFeedback**” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by sspeaks610** [comment]
I believe all breaking changes are false-positives.

6 of them are adding error responses when none were previously defined. This is due to the following s360 issue and some similar to it.

[INVALID_RESPONSE_CODE identified in operation Redis_ExportData of MICROSOFT.CACHE on API version 2019-07-01](https://portal.azure-devex-tools.com/amekpis/correctness/detail?errorId=6176E8EC-91E7-4EAA-80A4-111D451906D4)

Similarly, 2 of them are from endpoints that are really only used by the portal for user convenience and by the networking team calling our RP

[GET/SUBSCRIPTIONS/PROVIDERS/MICROSOFT.CACHE/LOCATIONS/ASYNCOPERATIONS/ on API version 2019-07-01](https://portal.azure-devex-tools.com/amekpis/completeness/detail?errorId=966A7E15-4E44-489D-AF1B-07246EE683A2)

File: specification/redis/resource-manager/Microsoft.Cache/preview/2019-07-01/redis.json [hunk 1]
```diff
@@ -1347,9 +1383,187 @@
...
+            "Failed",
+            "Canceled",
+            "InProgress"
+          ],
+          "x-ms-enum": {
+            "name": "OperationStatus",
+            "modelAsString": true
+          }
+        },
+        "startTime": {
```
Comment (filizt) [inline specification/redis/resource-manager/Microsoft.Cache/preview/2019-07-01/redis.json:1527]: please mark non-editable properties as readonly

File: specification/redis/resource-manager/Microsoft.Cache/preview/2019-07-01/redis.json [hunk 2]
```diff
@@ -1347,9 +1383,187 @@
...
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cache/locations/{location}/asyncOperations/{operationGuid}": {
```
Comment (majastrz) [inline specification/redis/resource-manager/Microsoft.Cache/preview/2019-07-01/redis.json:1440]: >asyncOperations/{operationGuid} [](start = 119, length = 31)

Most RPs don't model the async operations in the swagger. I'm curious why you're doing that.

File: specification/redis/resource-manager/Microsoft.Cache/preview/2019-07-01/redis.json [hunk 3]
```diff
@@ -1248,6 +1278,12 @@
         "responses": {
           "200": {
             "description": "Linked server was successfully deleted."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
```
Comment (majastrz) [inline specification/redis/resource-manager/Microsoft.Cache/preview/2019-07-01/redis.json:1285]: >ErrorResponse [](start = 98, length = 13)

The addition of `default` in this PR got flagged as breaking changes. This is a false positive and is not actually a breaking change.

File: specification/redis/resource-manager/Microsoft.Cache/preview/2019-07-01/redis.json [hunk 4]
```diff
@@ -1347,9 +1383,187 @@
           "nextLinkName": "nextLink"
         }
       }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Cache/redis/{cacheName}/privateLinkResources": {
```
Comment (majastrz) [inline specification/redis/resource-manager/Microsoft.Cache/preview/2019-07-01/redis.json:1387]: >privateLinkResources [](start = 116, length = 20)

Modification of an existing stable API version is not allowed anymore. (See [https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#versioning](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#versioning) for details.) You will need to follow the instructions in the list to start the breaking changes review.

**Comment by ChenTanyi** [comment]
> I believe all breaking changes are false-positives.
> 
> 6 of them are adding error responses when none were previously defined. This is due to the following s360 issue and some similar to it.
> 
> [INVALID_RESPONSE_CODE identified in operation Redis_ExportData of MICROSOFT.CACHE on API version 2019-07-01](https://portal.azure-devex-tools.com/amekpis/correctness/detail?errorId=6176E8EC-91E7-4EAA-80A4-111D451906D4)
> 
> https://msazure.visualstudio.com/Swagger%20KPIs/_workitems/edit/8187819
> 
> Similarly, 2 of them are from endpoints that are really only used by the portal for user convenience and by the networking team calling our RP
> 
> [GET/SUBSCRIPTIONS/PROVIDERS/MICROSOFT.CACHE/LOCATIONS/ASYNCOPERATIONS/ on API version 2019-07-01](https://portal.azure-devex-tools.com/amekpis/completeness/detail?errorId=966A7E15-4E44-489D-AF1B-07246EE683A2)
> 
> https://msazure.visualstudio.com/Swagger%20KPIs/_workitems/edit/8188012

@sspeaks610 Adding `new response` and `new API` both are considered as breaking change, according to @jianyexi. Please go through breaking change process.

**Comment by jianyexi** [comment]
> > I believe all breaking changes are false-positives.
> > 6 of them are adding error responses when none were previously defined. This is due to the following s360 issue and some similar to it.
> > [INVALID_RESPONSE_CODE identified in operation Redis_ExportData of MICROSOFT.CACHE on API version 2019-07-01](https://portal.azure-devex-tools.com/amekpis/correctness/detail?errorId=6176E8EC-91E7-4EAA-80A4-111D451906D4)
> > https://msazure.visualstudio.com/Swagger%20KPIs/_workitems/edit/8187819
> > Similarly, 2 of them are from endpoints that are really only used by the portal for user convenience and by the networking team calling our RP
> > [GET/SUBSCRIPTIONS/PROVIDERS/MICROSOFT.CACHE/LOCATIONS/ASYNCOPERATIONS/ on API version 2019-07-01](https://portal.azure-devex-tools.com/amekpis/completeness/detail?errorId=966A7E15-4E44-489D-AF1B-07246EE683A2)
> > https://msazure.visualstudio.com/Swagger%20KPIs/_workitems/edit/8188012
> 
> @sspeaks610 Adding `new response` and `new API` both are considered as breaking change, according to @jianyexi. Please go through breaking change process.

@sspeaks610 you can follow the action described at https://github.com/Azure/azure-rest-api-specs/pull/11737#issuecomment-729935983 , it's a quite simple process. According to your description , I think you can get approval .

**Comment by TimLovellSmith** [comment]
@sspeaks610 Sorry, did this get blocked on my review?

**Review by TimLovellSmith — COMMENTED** [review]
Added a few comments and upvoted the readonly one.

**Pull request closed without merging**